src/cursor-doc/model.ts

```diff
@@ -527,14 +538,19 @@ export class LineInputModel implements EditableModel {
 
 export class StringDocument implements EditableDocument {
   constructor(contents?: string) {
+    let lineEndingSeq = null;
+    if (contents) {
+      lineEndingSeq = getFirstLineEndingSeq(contents);
+    }
```

Comment:
I tend to prefer `const` over `let`.

```suggestion
    const lineEndingSeq = contents ? getFirstLineEndingSeq(contents) : null;
```

I think @bpringe has showed me a way to write that more succinctly, but I don't recall the syntax right now. But maybe chatGPT knows. 😄

Reply:
Updated.